From 17371d955d93f92ecd32ef1ac9b4aff4f5df034f Mon Sep 17 00:00:00 2001
From: Matthias Brugger <mbrugger@suse.com>
Date: Mon, 9 Nov 2015 11:09:06 +0100
Subject: [PATCH] arm64: numa: Convert WARN_ON to a pr_warn
Patch-mainline: Never, this is an overlay patch for the numa implementation. The problem should be fixed when the numa patches got accepted upstream.
References: fate#318444

If no numa dt binding is found, the numa code taints the kernel with a
WARN_ON. We degradate this to a normal warning as we don't want to taint
the kernel on all non-numa machines.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/mm/numa.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/mm/numa.c b/arch/arm64/mm/numa.c
index 0c879eb..7ff035d 100644
--- a/arch/arm64/mm/numa.c
+++ b/arch/arm64/mm/numa.c
@@ -446,8 +446,10 @@ static int __init numa_register_memblks(struct numa_meminfo *mi)
 	/* Account for nodes with cpus and no memory */
 	node_possible_map = numa_nodes_parsed;
 	numa_nodemask_from_meminfo(&node_possible_map, mi);
-	if (WARN_ON(nodes_empty(node_possible_map)))
+	if (nodes_empty(node_possible_map)) {
+		pr_warn("no node_possible_map found!");
 		return -EINVAL;
+	}
 
 	for (i = 0; i < mi->nr_blks; i++) {
 		struct numa_memblk *mb = &mi->blk[i];
-- 
2.1.4

